src/wallet/wallet2.cpp

```diff
@@ -13241,6 +13259,94 @@ size_t wallet2::import_outputs(const std::pair<uint64_t, std::vector<tools::wall
   return m_transfers.size();
 }
 //----------------------------------------------------------------------------------------------------
+size_t wallet2::import_outputs(const std::pair<uint64_t, std::vector<tools::wallet2::exported_transfer_details>> &outputs)
+{
+  PERF_TIMER(import_outputs);
+
+  THROW_WALLET_EXCEPTION_IF(outputs.first > m_transfers.size(), error::wallet_internal_error,
+      "Imported outputs omit more outputs that we know of");
+
+  const size_t offset = outputs.first;
+  const size_t original_size = m_transfers.size();
+  m_transfers.resize(offset + outputs.second.size());
+  for (size_t i = 0; i < offset; ++i)
+    m_transfers[i].m_key_image_request = false;
+  for (size_t i = 0; i < outputs.second.size(); ++i)
+  {
+    exported_transfer_details etd = outputs.second[i];
+    transfer_details &td = m_transfers[i + offset];
+
+    // skip those we've already imported, or which have different data
+    if (i + offset < original_size)
+    {
+      bool needs_processing = false;
+      if (!td.m_key_image_known)
+        needs_processing = true;
+#define CMPF(f) else if (!(etd.f == td.f)) needs_processing = true
+      CMPF(m_internal_output_index);
+      CMPF(m_global_output_index);
+#undef CMPF
+      else if (!(etd.m_pubkey == td.get_public_key()))
+        needs_processing = true;
+
+      // copy anyway, since the comparison does not include ancillary fields which may have changed
+      td = {};
```

Comment:
This overwrites `m_tx`, which is what was causing the seg fault later on in `export_key_images` that I was seeing [here](https://github.com/monero-project/monero/pull/8179/files#r808646275), since no more vouts. Annoyingly enough, it only happens when you re-import a 2nd time, which is why @r4v3r23 didn't see it in their tests

More importantly: because this block of code lives inside this `if (i + offset < original_size)`, it doesn't run on 1st import, and so the 1st import doesn't assign these values (most importantly, `m_global_output_index` doesn't get set on 1st import)

Based on my reading, seems to make sense to move this whole block of assignments right after `transfer_details &td = m_transfers[i + offset]` since it seems expected to always happen, and doesn't have much to do with `if (i + offset < original_size)`

...except for the `td = {}` assignment, don't think we want that because it just overwrites everything, can just get rid of that I think?